src/io/hdfs_filesys.cc

```diff
@@ -50,14 +50,25 @@ class HDFSStream : public SeekStream {
 
   virtual void Write(const void *ptr, size_t size) {
     const char *buf = reinterpret_cast<const char*>(ptr);
-    while (size != 0) {
-      tSize nwrite = hdfsWrite(fs_, fp_, buf, size);
-      if (nwrite == -1) {
+    size_t nleft = size;
```

Comment:
This line is redundant. Just use `size` is fine here.

Reply:
I prefer to keep it here to make it consistent with the `Read` implementation